patches/tModLoader/Terraria/ModLoader/Default/PurgeUnloadedCommand.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Linq;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader.IO;
+using System.Collections.Generic;
+using System;
+using Terraria.DataStructures;
+
+namespace Terraria.ModLoader.Default;
+
+internal class PurgeUnloadedCommand : ModCommand
+{
+	public override string Command => "purgeunloaded";
+	public override CommandType Type => CommandType.Chat; // complicated to implement correctly in MP
+	public override string Description => Language.GetTextValue("tModLoader.CommandPurgeUnloadedDescription");
+	public override string Usage => Language.GetTextValue("tModLoader.CommandPurgeUnloadedUsage");
+
+	public override void Action(CommandCaller caller, string input, string[] args)
+	{
+		if (Main.netMode != NetmodeID.SinglePlayer) {
+			caller.Reply("This command can only be called in Single Player mode.");
+			return;
+		}
+
+		bool purgeTiles = false;
+		bool purgeWalls = false;
+		bool actuallyPurge = false; // "dry run" otherwise
+
+		string modName = null;
+		for (int i = 0; i < args.Length; i++) {
+			if (args[i] == "-h") {
+				caller.Reply(Usage);
+				return;
+			}
+			else if (args[i] == "-t") {
+				purgeTiles = true;
+			}
+			else if (args[i] == "-w") {
+				purgeWalls = true;
+			}
+			// TODO: We can purge other content in the future as well with other flags
+			else if (args[i] == "-p") {
+				actuallyPurge = true;
+			}
+			else {
+				modName = args[i];
+			}
+		}
+
+		if (!purgeTiles && !purgeWalls) {
+			caller.Reply("At least one of '-w' or '-t' must be set");
+			caller.Reply(Usage);
+			return;
+		}
+
+		Dictionary<string, int> purgedTiles = [];
+		Dictionary<string, int> purgedWalls = [];
+		var unloadedTileTypes = TileIO.Tiles.unloadedTypes.ToHashSet();
+		var unloadedWallType = ModContent.WallType<UnloadedWall>();
+		List<Point16> coordinates = [];
+
+		for (int i = 0; i < Main.maxTilesX; i++) {
+			for (int j = 0; j < Main.maxTilesY; j++) {
+				Tile t = Main.tile[i, j];
+				if (purgeTiles && t.HasTile && t.TileType >= TileID.Count && unloadedTileTypes.Contains(t.TileType)) {
```

Comment:
This will be faster if you use an array than a `HashSet`, No need for the `t.TileType >=` check then either